pkg/apis/autoscaling/types.go

```diff
@@ -202,6 +202,62 @@ type HorizontalPodAutoscalerStatus struct {
 
 	// CurrentMetrics is the last read state of the metrics used by this autoscaler.
 	CurrentMetrics []MetricStatus
+
+	// conditions is the set of conditions required for this autoscaler to scale its target,
+	// and indicates whether or not those conditions are met.
+	Conditions []HorizontalPodAutoscalerCondition
+}
+
+// ConditionStatus indicates the status of a condition (true, false, or unknown).
+type ConditionStatus string
+
+// These are valid condition statuses. "ConditionTrue" means a resource is in the condition;
+// "ConditionFalse" means a resource is not in the condition; "ConditionUnknown" means kubernetes
+// can't decide if a resource is in the condition or not. In the future, we could add other
+// intermediate conditions, e.g. ConditionDegraded.
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+)
+
+// HorizontalPodAutoscalerConditionType are the valid conditions of
+// a HorizontalPodAutoscaler.
+type HorizontalPodAutoscalerConditionType string
+
+var (
+	// CanAccessScale indicates whether or not the HPA controller can access the scale endpoint
+	// of it's target.
+	CanAccessScale HorizontalPodAutoscalerConditionType = "CanAccessScale"
+	// InBackoff indicates that the HPA is current in backoff from another scale, and thus will
+	// not attempt new scaling.
+	InBackoff HorizontalPodAutoscalerConditionType = "InBackoff"
+	// CanComputeReplicas indicates whether or not the HPA controller can properly access metrics
+	// and compute replicas for the given HPA.
+	CanComputeReplicas HorizontalPodAutoscalerConditionType = "CanComputeReplicas"
```

Comment:
Although Available has specific meaning in the context of a deployment, which I don't want to overlay.  I don't think an HPA is necessarily available, since an HPA being available is independent of a deployment being available.

Active is as close a generic word as we have.  The HPA status is "Active" if the controller is able to get metrics, is observing the resources, etc.  The controller is "Active = false" for a variety of a reasons.  ScalingBackoff is probably reserved for a controller that is active but is unable to effectively scale.

Maybe this is roughly three types:

1. Will the HPA scale if necessary (Active or whatever)
2. Is the HPA unable to scale due to an error out of its control (Backoff / UnableToScale)
3. Is the HPA unable to scale due to being capped (ScaleLimited)

Those are all questions that I as a user would want to quickly select / alert on:

1. HPAs that aren't even configured in a sane way
2. HPAs exhibiting transient errors and thus may be undersized
3. HPAs that could grow more if I allowed it